src/NServiceBus.Core/Transports/Msmq/MsmqTransportInfrastructure.cs

```diff
@@ -33,17 +33,19 @@ public MsmqTransportInfrastructure(ReadOnlySettings settings, string connectionS
 
         ReceiveStrategy SelectReceiveStrategy(TransportTransactionMode minimumConsistencyGuarantee, TransactionOptions transactionOptions)
         {
-            if (minimumConsistencyGuarantee == TransportTransactionMode.TransactionScope)
+            switch (minimumConsistencyGuarantee)
             {
-                return new ReceiveWithTransactionScope(transactionOptions, new MsmqFailureInfoStorage(1000));
+                case TransportTransactionMode.TransactionScope:
```

Comment:
Would it be useful to rename the receive strategies to closer match the TransactionScope mode they support?
For example, something more like `TransactionScopeStrategy` or `SendsAtomicWithReceiveStrategy`.